xbmc/utils/StringUtils.cpp

```diff
@@ -336,76 +176,100 @@ int compareWchar (const void* a, const void* b)
   return 0;
 }
 
-wchar_t tolowerUnicode(const wchar_t& c)
+std::string transformToLower(std::string_view input, std::locale loc)
 {
-  wchar_t* p = (wchar_t*) bsearch (&c, unicode_uppers, sizeof(unicode_uppers) / sizeof(wchar_t), sizeof(wchar_t), compareWchar);
-  if (p)
-    return *(unicode_lowers + (p - unicode_uppers));
-
-  return c;
+  const std::ctype<char>& ct = use_facet<std::ctype<char> >(loc);
+  std::string output{input};;
+  transform(input.begin(), input.end(), output.begin(),
+      std::bind1st(std::mem_fun(&std::ctype<char>::tolower), &ct));
```

Comment:
`std::bind1st` isn't available in c++17

ref: https://en.cppreference.com/w/cpp/utility/functional/bind12

Reply:
`std::mem_fun` isn't available in c++17

https://en.cppreference.com/w/cpp/utility/functional/mem_fun